### ISSUE TITLE
Scalar plugin download links should work in Colab

### ISSUE THREAD

**Issue by wchargin** [tensorflow/tensorboard#1845]
Currently, the scalar plugin download links use `<a href download>`,
where the `href` points to a route on the TensorBoard backend. This
works fine locally, but on Colab requests to the backend are mediated by
a service worker, and the service worker does not appear to intercept
these requests. Consequently, downloads fail with a “network error”.

Proposed fix (to apply to Colab contexts only): When a run is selected
from the “run to download” dropdown menu, instead of updating the `href`
property of the links, send off two asynchronous `fetch` requests: one
to the CSV data and one to the JSON data. As these resolve, update the
`href`s to point to data URLs/blob URLs.

**Comment by nfelt** [comment]
FWIW, probably out of scope for now, but Stephan and I talked about moving the download data generation entirely to the client side so that there's no backend request required at all.  This would match the download-as-SVG behavior and let run selection directly drive the CSV/JSON downloaded data selection instead of having to pick specific runs in the download dropdown.

Originally mentioned in https://github.com/tensorflow/tensorboard/pull/1446#discussion_r221115827

**Comment by wchargin** [comment]
Interesting. That’s an attractive suggestion: hiding behind a clear
“download” button rather than a dropdown selection is nicer, and it
avoids the problem of the blob data going stale as more data is loaded
from the backend. It also means that we could get rid of the “Data
download links” button—one fewer UI option to care about. And, of
course, the user doesn’t have to reselect the run that they’ve already
presumably selected.

For the sake of getting Colab unbroken, I’m somewhat inclined to keep
the scope limited, but if this is something that we want to prioritize
then I’d be more than happy to shift gears to it.

**Comment by nfelt** [comment]
Yeah I agree that I don't think we should work on this right now.  But just FYI since it seems relevant.

**Comment by wchargin** [comment]
(Reopening: GitHub autoclosed this overzealously.)

**Comment by pindinagesh** [comment]
@wchargin 

Could you please move this to closed status as it is resolved. Thanks

**Comment by pindinagesh** [comment]
@wchargin 

Closing this issue for now, feel free to reopen if it's still persists and please have a look at similar issue which is described [here.](https://stackoverflow.com/questions/47818822/can-i-use-tensorboard-with-google-colab) Hope this helps. Thanks!

**Comment by nfelt** [comment]
This issue shouldn't be closed, it's still an active issue and was reported by a TensorBoard maintainer, not an end user.
